app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -231,17 +233,36 @@ provideSelectionArrow(
         if (node instanceof Ast.AutoscopedIdentifier) return node.identifier.id
         if (node instanceof Ast.PropertyAccess) return node.rhs.id
         if (node instanceof Ast.App) node = node.function
-        else break
+        else {
+          const wrapped = node.wrappedExpression()
+          if (wrapped != null) node = wrapped
+          else break
+        }
       }
       return null
     }),
     requestArrow: (target: RendererNode) => {
       arrowLocation.value = target
     },
     handled: false,
+    get suppressArrow() {
+      return suppressArrow.value
+    },
+    set suppressArrow(value) {
+      suppressArrow.value = value
+    },
   }),
 )
 
+watch(
+  () => isHovered.value && !suppressArrow.value,
```

Comment:
Putting this in a computed `showArrow` to use here and in the `v-if`/`v-else-if` would ensure that if our logic for when to show the arrow changes, it is still consistent with when we inhibit a parent arrow.